src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -59,26 +59,28 @@ public function onKernelRequest(RequestEvent $event)
             $exceptionListener->register($this->dispatcher);
         }
 
+        // Authentication listeners are pre-sorted by SortFirewallListenersPass
         $authenticationListeners = function () use ($authenticationListeners, $logoutListener) {
```

Comment:
Can't we completely remove lines 63 - 84 if they are already pre-sorted?

Reply:
Unfortunately not, since you still need to yield `$logoutListener`, which is not included in the `$authenticationListeners` array, at the correct position.

I have the feeling this is a historic leftover to ensure execution order for the logout listener. I believe nowadays the logout listener could easily be added to `$authenticationListeners` array and let the priority take care of execution order. Then we could in fact remove all of these lines here.

Maybe someone can confirm the reasoning why the `$logoutListener` is not provided together with `$authenticationListeners`.

Reply:
If you take a look at these two PRs, I think you're hypothesis is confirmed: https://github.com/symfony/symfony/pull/28100 + https://github.com/symfony/symfony/pull/28224

Maybe @chalasr can give a final confirmation though?

Reply:
Correct, yes. 
For the record, the logout listener was originally moved outside of the `$authenticationListeners` list in https://github.com/symfony/symfony/pull/24805.
As long as we make sure that the logout and access listeners are executed last, there should be no issue for removing this reordering logic and relying on the priority-based order. 
Existing tests should make sure we don't break things anyways